x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/AuthenticationService.java

```diff
@@ -379,7 +380,8 @@ private void consumeUser(User user, Map<Realm, Tuple<String, Exception>> message
          * names of users that exist using a timing attack
          */
         private void lookupRunAsUser(final User user, String runAsUsername, Consumer<User> userConsumer) {
-            final List<Realm> realmsList = realms.asList();
+            final List<Realm> realmsList =
+                    realms.asList().stream().filter((realm) -> realm.hasLookupDependency() == false).collect(Collectors.toList());
```

Comment:
I think this change only exists because your made that change to `lookupUser` to delegate to `lookup_realms`; if we drop that change then we can drop this too.